nextest-runner/src/cargo_config.rs

```diff
@@ -498,6 +515,8 @@ pub(crate) struct CargoConfig {
     #[serde(default)]
     pub(crate) build: CargoConfigBuild,
     pub(crate) target: Option<BTreeMap<String, CargoConfigRunner>>,
+    #[serde(default)]
+    pub(crate) env: BTreeMap<String, String>,
```

Comment:
This needs to support the full spec, with `force` and `relative` (and also needs to include tests for all the possible combinations).

Reply:
How do you recommend that I structure the tests? I'm not sure how to write an integration test that verifies that the environment variables are being set.

Reply:
That is a really good question! You'd do it through adding assertions to the `test_cargo_env_vars` test fixture: https://github.com/nextest-rs/nextest/blob/main/fixtures/nextest-tests/tests/basic.rs#L117. You can add an `[env]` section to https://github.com/nextest-rs/nextest/blob/main/fixtures/nextest-tests/.cargo/config to define environment variables.

So we'd have to introduce a bunch of assertions:
1. Plain env var that doesn't already exist: pick a name that no one's going to set like `__NEXTEST_PLAIN_ENV_DOES_NOT_EXIST` and set it to some sort of well-known value e.g. `plain-env-value`, then assert in the fixture that it's the same.
2. Plain env var that already exists: define a variable, and then also set it in https://github.com/nextest-rs/nextest/blob/main/nextest-runner/tests/integration/fixtures.rs#L200-L204 (rename `set_rustflags` to `set_env_vars` first) and https://github.com/nextest-rs/nextest/blob/515eef9f85a517a688dfde6946a8fcdbb4af9192/cargo-nextest/src/tests_integration/fixtures.rs#L131.
3. `force = true`: same as 2, except make sure it's set to the new value
4. `relative = true`
5. `relative = true, force = true`